choreolib/src/main/java/choreo/auto/AutoChooser.java

```diff
@@ -176,4 +222,21 @@ public void addAutoRoutine(String name, AutoRoutineGenerator generator) {
   public AutoRoutine getSelectedAutoRoutine() {
     return lastAutoRoutine;
   }
+
+  /**
+   * Gets a Command that runs the latest selected auto routine,
+   * stopping when the selected routine finishes.
+   * This Command can directly be bound to a trigger, like so:
+   * <pre><code>
+   *     AutoChooser chooser = ...;
+   *
+   *     public Robot() {
+   *         autonomous().whileTrue(chooser.autoCmd());
+   *     }
+   * </code></pre>
+   * @return A command that runs the selected {@link AutoRoutine}
+   */
+  public Command autoCmd() {
+    return Commands.defer(() -> getSelectedAutoRoutine().cmd(), Set.of());
```

Comment:
This behaviour is wrong, it erases requirements.
```suggestion
    return Commands.defer(() -> getSelectedAutoRoutine().cmd().asProxy(), Set.of());
```

Reply:
the auto routine command doesn't have any requirements; they are allocated when the triggers activate.

Reply:
ah yes that makes sense. add a comment above here explaining that